akka-docs/rst/scala/persistence-schema-evolution.rst

```diff
@@ -0,0 +1,386 @@
+  and its performance is also not very high (it never was designed for high-throughput scenarios).
+
+.. _Google Protocol Buffers: https://developers.google.com/protocol-buffers/
+.. _Apache Thrift: https://thrift.apache.org/
+
+Configuring payload serializers
+-------------------------------
+This section aims to highlight the complete basics on how to define custom serializers using :ref:`Akka Serialization <serialization-scala>`.
+Many journal plugin implementations use Akka Serialization, thus it is tremendously important to understand how to configure
+it to work with your event classes.
+
+.. note::
+  Read the :ref:`Akka Serialization <serialization-scala>` docs to learn more about defining custom serializers,
+  to improve performance and maintainability of your system. Do not depend on Java serialization for production deployments.
+
+The below snippet explains in the minimal amount of lines how a custom serializer can be registered.
+For more in-depth explanations on how serialization picks the serializer to use etc, please refer to its documentation.
+
+First we start by defining our domain model class, here representing a person:
+.. includecode:: code/docs/persistence/PersistenceSchemaEvolutionDocSpec.scala#simplest-custom-serializer-model
+
+Next we implement a serializer (or extend an existing one to be able to handle the new ``Person`` class):
+
+.. includecode:: code/docs/persistence/PersistenceSchemaEvolutionDocSpec.scala#simplest-custom-serializer
+
+And finally we register the serializer and bind it to handle the ``docs.persistence.Person`` class:
+
+.. includecode:: code/docs/persistence/PersistenceSchemaEvolutionDocSpec.scala#simplest-custom-serializer-config
+
+Deserialization will be performed by the same serializer which serialized the message initially
+because of the ``identifier`` being stored together with the message.
+
+For more advanced techniques for using serializers refer to the :ref:`Akka Serialization <serialization-scala>` documentation,
+giving the :ref:`string-manifest-serializer-scala` special attention, as it is tremendously useful for Persistence based applications
+(as we will see in some of the below explained evolutions).
+
+
+Serializing to human-readable documents
+---------------------------------------
+Some projects may prefer to store events in human-readable formats instead of formats optimised for write/read throughput.
+The style of serialization used in Akka Persistence is completely left up to the concrete Journal plugin implementations -
+this means that some community provided journals may offer human readable serialization, however it is not the default style
+to persist events.
+
+Additionally some datastores are able to store "documents" as JSON (or BSON) natively, wich may have additional benefits
+for later debugging or searchability through the events. In general we highly encourage the use projections using
```

Comment:
strange sentence, perhaps "In general we highly encourage using :ref:`akka-persistence-query-scala` to migrate..."